graph/src/data_source/subgraph.rs

```diff
@@ -354,18 +354,38 @@ pub struct MappingEntityTrigger {
 pub struct TriggerData {
     pub source: DeploymentHash,
     pub entity: EntitySourceOperation,
+    pub source_idx: u32,
 }
 
 impl TriggerData {
-    pub fn new(source: DeploymentHash, entity: EntitySourceOperation) -> Self {
-        Self { source, entity }
+    pub fn new(source: DeploymentHash, entity: EntitySourceOperation, source_idx: u32) -> Self {
+        Self {
+            source,
+            entity,
+            source_idx,
+        }
     }
 
     pub fn entity_type(&self) -> &str {
         self.entity.entity_type.as_str()
     }
 }
 
+impl Ord for TriggerData {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        match self.source_idx.cmp(&other.source_idx) {
```

Comment:
This is somewhat important, I think would be worth some comments/documentation about the decision and why it is working this way. Also probably a simple test to ensure this assumption is not broken going forward